src/object-observer.js

```diff
@@ -93,7 +97,14 @@ const
 		let l = observers.length;
 		while (l--) {
 			try {
-				observers[l](changes);
+				const target = observers[l]
+				const { path } = target.options
+
+				if (path && changes.every((change) => change.path.join('.').startsWith(path))) {
```

Comment:
:) sorry to bother you, yet it is still or all or nothing, which is not as the API is meant to be (deliver those and only those changes that are relevant)
what do you think about this:
```javascript
if (path) {
   let relevantChanges = changes.filter(change => change.path.join('.').startsWith(path));
   if (relevantChanges.length) {
      target.observer(relevantChanges);
   }
} else {
   target.observer(changes);
}
```

Reply:
Hei, no worries getting feedback is always good, but i do have 1 question still.

```JS
let relevantChanges = changes.every(change => change.path.join('.').startsWith(path));
```

If i would use `every` instead of `filter`, it seems to work the same, is there any reason i should prefer to use `filter` there ?

The syntax just feels a little bit cleaner when using `every` in my opinion.

Reply:
What if you have 3 changes and only 2 of them meeting the conditional path.
In your version, you'll get `false` n the `relevantChanges` and none of them will be delivered to the recipient, right?
On the other hand, `filter` based version will have a 2 changes in the `relevantChanges` array, and those will be delivered as expected.